File: app/src/main/java/com/example/android/tj/model/TJServiceCommand.java
package com.example.android.tj.model;

import android.support.annotation.NonNull;

import com.google.gson.Gson;

public class TJServiceCommand {
    public int cmdCode;
    public int arg1;

    public String data;

    public TJServiceCommand(int cmdCode) {
        this.cmdCode = cmdCode;
    }

    public TJServiceCommand(int cmdCode, int arg1) {
        this.cmdCode = cmdCode;
        this.arg1 = arg1;
    }

    public TJServiceCommand(int cmdCode, String data) {
        this.cmdCode = cmdCode;
        this.data = data;
    }

    @NonNull
    public String toString() {
        return new Gson().toJson(this);
    }

    public static TJServiceCommand fromJson(String jsonStr) {
        return new Gson().fromJson(jsonStr, TJServiceCommand.class);
    }
}

File: app/src/main/java/com/example/android/tj/model/TJServiceCommand.kt
package com.example.android.tj.model

import com.google.gson.Gson

class TJServiceCommand {
    var cmdCode: Int = 0
    var arg1: Int = 0

    var data: String = ""

    constructor(cmdCode: Int) {
        this.cmdCode = cmdCode
    }

    //TODO: more general representation for arg1
    // polymorphism?
    constructor(cmdCode: Int, arg1: Int) {
        this.cmdCode = cmdCode
        this.arg1 = arg1
    }

    constructor(cmdCode: Int, data: String) {
        this.cmdCode = cmdCode
        this.data = data
    }

    override fun toString(): String {
        return Gson().toJson(this)
    }

    companion object {

        fun fromJson(jsonStr: String): TJServiceCommand {
            return Gson().fromJson(jsonStr, TJServiceCommand::class.java)
        }
    }
}
